### PR TITLE
Require Node.js 8 and ESLint 6

### PR DESCRIPTION
**Pull request by scttcper**
updated everything to the latest versions let me know if you would rather i match eslint-config-xo https://github.com/xojs/eslint-config-xo/blob/master/package.json#L55-L62

Currently, this package warns peerdependency when using eslint 6

Adds testing in node 12 and removes 6

BREAKING CHANGE: changes the minimum node version to 8 in package.json

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-import tempWrite from 'temp-write';
+import path from 'path';
 
 const fixture = `'use strict';\nconst x = true;\n\nif (x) {\n  console.log();\n}\n`;
 
 function runEslint(str, conf) {
 	const linter = new eslint.CLIEngine({
 		useEslintrc: false,
-		configFile: tempWrite.sync(JSON.stringify(conf))
+		configFile: path.join(__dirname, conf)
```
Comment (sindresorhus) [inline test/test.js]: Why this change? It was more elegant when you could just pass the config as an object.

Comment (scttcper) [reply test/test.js]: I really wasn't able to get that to work on eslint 6. Because the file created was located in the "temp" directory `/temp/something` it couldn't find the node_modules folder to use the eslint configs listed under extends.

**Comment by sindresorhus** [comment]
Thanks :)
